frame/support/src/storage/mod.rs

```diff
@@ -1133,7 +1133,11 @@ pub trait StoragePrefixedMap<Value: FullCodec> {
 		crate::storage::storage_prefix(Self::module_prefix(), Self::storage_prefix())
 	}
 
-	/// Remove all value of the storage.
+	/// Remove all values of the storage in the overlay and up to `limit` in the backend.
+	///
+	/// All values in the client overlay will be deleted, if there is some `limit` then up to
```

Comment:
dq: is the overlay just a cache?

Reply:
I realized with this PR that the sp_io::storage::clear function is leaking the concept of overlay into the API, which doesn't seems the best.

It is more a cache of the written key/value, all key/value written from the beginning of the block will be stored in the overlay, and then at the end of the block import all the changes are committed.